mypy/config_parser.py

```diff
@@ -322,21 +335,8 @@ def parse_config_file(
     else:
         section = parser["mypy"]
 
-        if "files" in section:
-            raw_files = section["files"].strip()
-            files_split = [file.strip() for file in raw_files.split(",")]
-
-            # Remove trailing empty entry if present
-            if files_split and files_split[-1] == "":
-                files_split.pop()
-
-            # Raise an error if there are any remaining empty strings
-            if "" in files_split:
-                raise ValueError(
-                    "Invalid config: Empty filenames are not allowed except for trailing commas."
-                )
-
-            options.files = files_split
+        if "files" in section and isinstance(raw_files := section["files"], str):
```

Comment:
Is this only a string in the case of INI configs? Does TOML not hit this code path? Would the string check be unnecessary when it's known it's going through the INI code path?

It seems to me that a better place for injecting this conversion would be inside `_parse_individual_file()` where it has the differentiation between `is_toml()` and not. And you'd only apply it in the else-branch there.

This would likely let you drop the str check for good.